### PR TITLE
[KissmangaORG] add website support

### LINKED ISSUE #2237: [Site Request] Kissmanga
**Name of the website**
KissManga

**Website urls (examples below)**
- Site: https://kissmanga.org/
- Manga List: hhttps://kissmanga.org/manga_list
- Manga example: https://kissmanga.org/manga/azur_lane_4koma_slow_ahead
- Chapter online viewer example: https://kissmanga.org/chapter/azur_lane_4koma_slow_ahead/chapter_57

**Languages**
List of languages supported by the website
-English

**Website relationship*
If applicable describe the relation with any other website (eg : "alternative domain, copy of ...").

**Additional details**
Add any other context details that you may have found (like template to reuse)

### LINKED ISSUE #2591: [Site Request] Kissmanga.org
**Name of the website**
Kissmanga

**Website urls (examples below)**
- Site: https://kissmanga.org/
- Manga List: https://kissmanga.org/manga_list
- Manga example: https://kissmanga.org/manga/my_id_is_gangnam_beauty
- Chapter online viewer example: https://kissmanga.org/chapter/my_id_is_gangnam_beauty/chapter_91

**Languages**
List of languages supported by the website
English

**Website relationship*
If applicable describe the relation with any other website (eg : "alternative domain, copy of ...").

**Additional details**
Add any other context details that you may have found (like template to reuse)

### PR DESCRIPTION
**Pull request by manh21**
Resolve #2237 
References: https://github.com/fmd-project-team/FMD/pull/1141
References: https://github.com/dazedcat19/FMD2/pull/139

### CODE REVIEW

File: src/web/mjs/connectors/KissmangaORG.mjs [hunk 1]
```diff
@@ -0,0 +1,69 @@
+import Connector from '../engine/Connector.mjs';
+import Manga from '../engine/Manga.mjs';
+
+export default class KissmangaORG extends Connector {
+
+    constructor() {
+        super();
+        super.id = 'kissmangaorg';
+        super.label = 'Kissmanga.org';
+        this.tags = [ '!test', 'manga', 'webtoon', 'english' ];
```
Comment (27136773) [inline src/web/mjs/connectors/KissmangaORG.mjs]: Is the `test` tag intentional?

File: src/web/mjs/connectors/KissmangaORG.mjs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+    async _getChapters(manga) {
+        const uri = new URL(manga.id, this.url);
+        const request = new Request(uri, this.requestOptions);
+        const data = await this.fetchDOM(request, this.queryChapters);
+        return data.map(element => {
+            return {
+                id: this.getRootRelativeOrAbsoluteLink(element, this.url),
+                title: element.text.replace('-', '').replace(manga.title, '').trim()
```
Comment (27136773) [inline src/web/mjs/connectors/KissmangaORG.mjs]: Is it safe to assume `-` can always be removed and does not appear as part of the chapter name?
Is it intentional that only the first occurrence of `-` will be removed?

File: src/web/mjs/connectors/KissmangaORG.mjs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+            };
+        });
+    }
+
+    async _getPages(manga) {
+        const uri = new URL(manga.id, this.url);
+        const request = new Request(uri, this.requestOptions);
+        const data = await this.fetchDOM(request, this.queryPages);
+        return data.map(element => this.getAbsolutePath(element.dataset['src'] || element.dataset['data-src'] || element.src, this.url)).filter(link => !link.includes('adskeeper.co.uk'));
```
Comment (27136773) [inline src/web/mjs/connectors/KissmangaORG.mjs]: Do the images have a dataset attribute at all?
Are there any `adskeeper` image links on this website that needs to be filtered out?

**Review by 27136773 — APPROVED** [review]
Looks good

File: src/web/mjs/connectors/KissmangaORG.mjs (hunk 1 above)
Comment (manh21) [reply src/web/mjs/connectors/KissmangaORG.mjs]: development artifact

**Comment by manh21** [comment]
Ready for merge

**Review by 27136773 — APPROVED** [review]
Looks good

**Pull request merged**